File: Anul II/Medii de proiectare si programare/Laborator/lab09/persistence/src/main/java/com/ubb/repository/IRaceEntryRepository.java
package com.ubb.repository;

import com.ubb.model.Race;
import com.ubb.model.RaceEntry;

import java.util.Collection;

public interface IRaceEntryRepository extends IRepository<Long, RaceEntry> {
    Collection<RaceEntry> getEntriesByRace(Long raceID);

    void deleteByIds(Long participantID, Long raceID);

    Collection<Race> getRacesWhereNotRegisteredAndEngineCapacity(Long participantID, Integer engineCc);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Repository/src/IRaceEntryRepository.cs
using System.Collections;
using Ubb.BikeContest.Model;

namespace Ubb.BikeContest.Repository;

public interface IRaceEntryRepository : IRepository<long, RaceEntry>
{
    IEnumerable<RaceEntry> GetEntriesByRace(long raceId);

    IEnumerable<Race> GetRacesWhereNotRegisteredAndEngineCapacity(long participantId, int engineCapacity);
}